### PR TITLE
Add logger configuration to allow prod exception handling

### PR DESCRIPTION
**Pull request by AetherUnbound**
## Description

This PR adds some additional logging configuration to enable exception logging in production. Currently, when an exception or error occurs, we get unhelpful logs that look like this:

```
2022-01-07T13:06:57.964-08:00	Internal Server Error: /v1/audio/
2022-01-07T13:06:57.965-08:00	Internal Server Error: /v1/audio/
2022-01-07T13:06:57.984-08:00	Internal Server Error: /v1/audio/
2022-01-07T13:06:57.985-08:00	Internal Server Error: /v1/audio/
```

The above errors are HTTP 500 errors, but we aren't able to see any information about what is actually failing. The logging configuration in this PR ensures that:

1. Intentional logging within the service gets bubbled up appropriately
1. Django logging in general is set to `WARNING` in production
1. Exceptions are bubbled up to the logs appropriately in production

This does _not_ mean that users will see the debug view that Django provides in production. When we have `DEBUG=False`, all of that will be correctly hidden. This just makes it easier for us to troubleshoot production problems (particularly ones we can't replicate locally!).

I've used this config previously with great success! We also have the option to set up admin emails down the line where we'll receive an email on server errors. I think for a project of this size, that might be inadvisable and we would want a different way of aggregating/reporting errors.

## Testing Instructions

_Note that these steps are not to be committed and are only intended to replicate a production setup locally_
1. Set `DJANGO_DEBUG_ENABLED=False` in `api/env.docker`
1. Set the condition on line 54 of `settings.py` from `if DEBUG:` to `if True:`
1. Add an exception under the healthcheck view in `api/catalog/api/views/health_views.py` (I just added `raise ValueError("Whoops!")` above line 17)
1. `just up`
1. `just logs web`
1. Visit localhost:8000/healthcheck in your browser

If you follow these steps and are on `main`, you should receive an HTTP 500 error page and there should be nothing printed out in the logs (or at least not an exception). However if these steps are done on this branch, you should still receive the same generic HTTP 500 page in the browser, but the logs should contain the full exception traceback.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the *default* branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Review by obulat — APPROVED** [review]
Should be really helpful! 
Thank you for the detailed description and testing instructions, it's really interesting to read.

**Comment by AetherUnbound** [comment]
Ugh, this is suffering the same hanging issue as I tried to address in #421 :/ I'll need to investigate this a bit more

File: api/catalog/settings.py [hunk 1]
```diff
@@ -17,6 +17,8 @@
 from decouple import config
 from sentry_sdk.integrations.django import DjangoIntegration
 
+from .logger import LOGGING  # noqa: F401
```
Comment (zackkrida) [inline api/catalog/settings.py:20]: `# noqa: F401`

What does this mean?

File: api/catalog/logger.py [hunk 2]
```diff
@@ -0,0 +1,82 @@
+from logging import LogRecord
+
+
+def health_check_filter(record: LogRecord) -> bool:
+    # Filter out health checks from the logs, they're verbose and happen frequently
+    return not ("GET /healthcheck" in record.getMessage() and record.status_code == 200)
```
Comment (zackkrida) [inline api/catalog/logger.py:6]: ❤️

**Review by zackkrida — APPROVED** [review]
Looks nice! @rbadillap I just want to point out that you were an assigned reviewer here; in the future just throw a comment on someone else's PR if your not able to review in within a day or two and we'll assign someone else. Thanks!

File: api/catalog/settings.py (hunk 1 above)
Comment (AetherUnbound) [reply api/catalog/settings.py:20]: This `LOGGING` variable is used within `settings.py` and is expected to be present when Django tries to access `settings.LOGGING`. However we don't use it anywhere else in the file, so our linter complains that it's an unnecessary import. This `noqa` disables that particular check (`F401`) for this line.

Comment (zackkrida) [reply api/catalog/settings.py:20]: Thanks!

**Pull request merged**